### Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):  
### zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],  
### ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].
### Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy. Полученные значения должны быть равны. Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков, а затем с использованием функций из библиотек numpy и pandas.

In [ ]:
import numpy as np

zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
kz = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

#посчитаем среднее каждого списка
Mzp = zp.mean()
Mkz = kz.mean()

cov = ((zp - Mzp) * (kz - Mkz)).sum() / (len(zp) - 1)
print(cov)

10175.37777777778


Теперь при помощи cov

In [ ]:
print(np.cov(zp, kz))

[[ 3882.93333333 10175.37777778]
 [10175.37777778 33854.32222222]]


Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков

$$r_{XY} = \frac{cov_{XY}}{\sigma_{X}\sigma_{Y}}$$

In [ ]:
corr = ((zp - Mzp) * (kz - Mkz)).sum() / (len(zp)) / (zp.std() * kz.std())
corr

0.8874900920739162

При помощи corrcoef

In [ ]:
corrcoef = np.corrcoef(zp, kz)
corrcoef

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

### Измерены значения IQ выборки студентов, обучающихся в местных технических вузах: 131, 125, 115, 122, 131, 115, 107, 99, 125, 111.  Известно, что в генеральной совокупности IQ распределен нормально.  Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [ ]:
iq = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])
iq_mean = iq.mean()
iq_s = iq.std(ddof=1)
print(iq_mean, iq_s)

118.1 10.54566788359614


$$
P \left( 
118.1 - t_{1 - 0.95/2, \: 10 - 1} \cdot \dfrac{10.5456}{\sqrt{10}} 
\leq M(X) \leq
118.1 + t_{1 - 0.95/2, \: 10 - 1} \cdot \dfrac{10.5456}{\sqrt{10}} 
\right) = 1 - \alpha.
$$

$$
P \left( 
118.1 - 2.2621 \cdot \dfrac{10.5456}{3,1622} 
\leq M(X) \leq
118.1 + 2.2621\cdot \dfrac{10.5456}{3,1622} 
\right) = 1 - \alpha.
$$

$$
P \left( 
118.1 - 7,5438 
\leq M(X) \leq
118.1 + 7,5438 
\right)
$$

$$
 \left( 
110.55 
\leq M(X) \leq
125.64
\right)
$$

### Известно, что рост футболистов в сборной распределен нормально с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27, среднее выборочное составляет 174.2. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [ ]:
#получим среднеквадратическое отклонение взяв корень из дисперсии.
s = np.sqrt(25) #5

$$
P \left( 
174.2 - t_{1 - 0.95/2, \: 27 - 1} \cdot \dfrac{5}{\sqrt{27}} 
\leq M(X) \leq
174.2 + t_{1 - 0.95/2, \: 27 - 1} \cdot \dfrac{5}{\sqrt{27}} 
\right) = 1 - \alpha.
$$

$$
 \left( 
174.2 - 2.0555 \cdot \dfrac{5}{\sqrt{27}} 
\leq M(X) \leq
174.2 + 2.0555 \cdot \dfrac{5}{\sqrt{27}} 
\right)
$$

$$
 \left( 
174.2 - 1,9779
\leq M(X) \leq
174.2 + 1,9779
\right)
$$

$$
 \left( 
172,22
\leq M(X) \leq
176,17
\right)
$$